--- frontend/coprs_frontend/coprs/context_processors.py	2023-01-29 11:52:20.854449497 +0800
+++ ../copr_docker/docker/frontend/files/usr/share/copr/coprs_frontend/coprs/context_processors.py	2023-03-09 15:57:41.500076207 +0800
@@ -67,6 +67,12 @@
                 'desc': 'sign up',
             })
 
+        if config['OIDC_LOGIN'] and app.config['OIDC_PROVIDER_NAME']:
+            menu.append({
+                'link': flask.url_for("misc.oidc_login"),
+                'desc': '{} login'.format(app.config['OIDC_PROVIDER_NAME']),
+            })
+
     return dict(login_menu=menu)
 
 @app.context_processor
