diff --git a/frontend/coprs_frontend/application b/frontend/coprs_frontend/application
index 753add041..2a8d28ded 100644
--- a/frontend/coprs_frontend/application
+++ b/frontend/coprs_frontend/application
@@ -9,6 +9,14 @@ os.environ["COPRS_ENVIRON_PRODUCTION"] = "1"
 sys.path.insert(0, os.path.dirname(__file__))
 
 from coprs import app
+from werkzeug.middleware.proxy_fix import ProxyFix
+
+app.wsgi_app = ProxyFix(
+    app.wsgi_app, x_for=1, x_proto=1, x_host=1, x_prefix=1
+)
+
+from werkzeug import Request
+Request.max_form_parts = 5000 # fix /rebuild-all failed when submitted more than 1000 packages
 
 if app.debug:
     from werkzeug.debug import DebuggedApplication
