diff --git a/frontend/coprs_frontend/coprs/views/misc.py b/frontend/coprs_frontend/coprs/views/misc.py
index 0f0c4f76a..217d7209a 100644
--- a/frontend/coprs_frontend/coprs/views/misc.py
+++ b/frontend/coprs_frontend/coprs/views/misc.py
@@ -200,7 +200,7 @@ def login_required(role=RoleEnum("user")):
         @functools.wraps(f)
         def decorated_function(*args, **kwargs):
             if flask.g.user is None:
-                return flask.redirect(flask.url_for("misc.oid_login",
+                return flask.redirect(flask.url_for("misc.oidc_login",
                                                     next=flask.request.url))
 
             if role == RoleEnum("admin") and not flask.g.user.admin:
