diff --git a/backend/run/copr_fix_gpg.py b/backend/run/copr_fix_gpg.py
index e5b124c95..24edf74ef 100755
--- a/backend/run/copr_fix_gpg.py
+++ b/backend/run/copr_fix_gpg.py
@@ -70,11 +70,11 @@ def fix_copr(args, opts, copr_full_name):
         return
 
     log.info("Generate key-pair on copr-keygen (if not generated) for email %s",
-             create_gpg_email(owner, coprname))
+             create_gpg_email(owner, coprname, opts.sign_domain))
     create_user_keys(owner, coprname, opts)
 
     log.info("Regenerate pubkey.gpg in copr %s", copr_path)
-    get_pubkey(owner, coprname, log, os.path.join(copr_path, 'pubkey.gpg'))
+    get_pubkey(owner, coprname, log, opts.sign_domain, os.path.join(copr_path, 'pubkey.gpg'))
 
     # Match the "00001231-anycharacer" directory names.  Compile once, use many.
     builddir_matcher = re.compile(r"\d{8,}-")
diff --git a/backend/run/copr_sign_unsigned.py b/backend/run/copr_sign_unsigned.py
index 23fb557b9..33eaeae35 100755
--- a/backend/run/copr_sign_unsigned.py
+++ b/backend/run/copr_sign_unsigned.py
@@ -82,7 +82,7 @@ def check_pubkey(pubkey_path, user, project, opts):
     else:
         log.info("Missing pubkey for %s/%s", user, project)
         try:
-            get_pubkey(user, project, log, pubkey_path)
+            get_pubkey(user, project, log, opts.sign_domain, pubkey_path)
             return True
         except Exception as err:
             log.exception(err)
@@ -116,7 +116,7 @@ def main():
             log.info("Checking project dir: %s", project_name)
 
             try:
-                get_pubkey(user_name, project_name, log)
+                get_pubkey(user_name, project_name, log, opts.sign_domain)
                 log.info("Key-pair exists for %s/%s", user_name, project_name)
             except CoprSignNoKeyError:
                 create_user_keys(user_name, project_name, opts)
